cbench/fakeswitch.c

```diff
@@ -88,18 +88,22 @@ void fakeswitch_learn_dstmac(struct fakeswitch *fs)
 {
     // thanks wireshark
     char gratuitous_arp_reply [] = {
-        0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x0c, 
-        0x29, 0x1a, 0x29, 0x1a, 0x08, 0x06, 0x00, 0x01, 
-        0x08, 0x00, 0x06, 0x04, 0x00, 0x02, 0x00, 0x0c, 
-        0x29, 0x1a, 0x29, 0x1a, 0x7f, 0x00, 0x00, 0x01, 
+        0xff, 0xff, 0xff, 0xff, 0xff, 0xff, 0x00, 0x0c, // 8 octects per line
+                                // Ethernet Type: 802.1Q VLAN tagging (0x8100)
+        0x29, 0x1a, 0x29, 0x1a, 0x81, 0x00, 0x00, 0x01, //<- this 0x00, 0x01 (which is 1) is the vlan_id
+        // Ethernet Type: ARP (0x0806) 
+        0x08, 0x06, 0x06, 0x04, 0x00, 0x02, 0x00, 0x0c, 
+        // ARP OP Code: 1 is request, 2 is reply
+        0x00, 0x01, 0x29, 0x1a, 0x7f, 0x00, 0x00, 0x01, 
```

Comment:
Is this part fixed or tunable?

Reply:
This part we can customize it according to our future needs.

At the moment, the src ip(`ip_address_to_learn`) and destination ip(`destination_ip_address`) can be tuned.